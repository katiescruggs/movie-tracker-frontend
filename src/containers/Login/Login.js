import React, { Component } from 'react';
import { withRouter, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import { setCurrentUser } from '../../actions';
import './Login.css'
import * as actions from '../../actions';

class Login extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      email: '',
      password: '',
      showRegister: false
    };
  }

  handleInputChange(event, prop) {
    const value = event.target.value;
    this.setState({[prop]: value});
  }

  submitLogin = async (payload) => {
    if(this.props.showRegister) {
      this.props.userSignupAttempt(payload);
    } else {
      this.props.userLoginAttempt(payload);
    }
  }

  render() {
    const {name, email, password} = this.state;


    if (this.props.showRegister) {
      return (
        <div className='login'>
          <input className='input-field' onChange={(event) => this.handleInputChange(event, 'name')} type="text" placeholder="Name" />
          <input className='input-field'onChange={(event) => this.handleInputChange(event, 'email')} type="email" placeholder="email" />
          <input className='input-field'onChange={(event) => this.handleInputChange(event, 'password')} type="password" placeholder="password" />
          <button className='btn-submit'onClick={(event) => this.submitLogin({name, email, password})}>
            Submit
          </button>
        </div>
      );
    };
    return (
      <div className='login'>
        <input className='input-field' onChange={(event) => this.handleInputChange(event, 'email')} type="email" placeholder="email" />
        <input className='input-field' onChange={(event) => this.handleInputChange(event, 'password')} type="password" placeholder="password" />
        <button className='btn-submit' onClick={(event) => this.submitLogin({email, password})}>
          Submit
        </button>
      </div>
      );
  }
}

const mapStateToProps = state => {
  return {

  };
};

const mapDispatchToProps = dispatch => {
  return {
    userLoginAttempt: (userInfo) => {
      dispatch(actions.userLoginAttempt(userInfo));
    },
    userSignupAttempt: (userInfo) => {
      dispatch(actions.userSignupAttempt(userInfo));
    },
    getCurrentUser: (userInfo) => {
      dispatch(actions.getCurrentUser(userInfo));
    },
    userLogout: (userInfo) => {
      dispatch(actions.userLogout(userInfo));
    },
  };
};

export default withRouter(connect(null, mapDispatchToProps)(Login));


