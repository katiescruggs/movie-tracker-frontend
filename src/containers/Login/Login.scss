.login {
  background: lightblue;
  display: flex;
  height: 200px;
  width: 400px;
  position: fixed;
  top: calc(50% - 100px);
  left: calc(50% - 200px);
  flex-direction: column;
  justify-content: center;
  align-items: center;
  z-index: 1;
}

.input-field,
.btn-submit {
  margin-top: 20px;
}