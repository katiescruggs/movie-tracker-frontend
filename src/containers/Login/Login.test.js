import React from 'react';
import { shallow } from 'enzyme';
import { Login, mapStateToProps, mapDispatchToProps } from './Login';

describe('Login Container', () => {
  describe('Login', () => {
    let login;
    const mockLoginAttempt = jest.fn();
    const mockSignupAttempt = jest.fn();
    const mockGetCurrentUser = jest.fn();

    beforeEach( () => {
      login = shallow(
        <Login 
          name='Katie'
          showRegister={true}
          userLoginAttempt={mockLoginAttempt}
          userSignupAttempt={mockSignupAttempt}
          getCurrentUser={mockGetCurrentUser}
          errorStatus={false}
          errorMessage='an error message here'
        />);
    });

    it('should exist', () => {
      expect(login).toBeDefined();
    });

    it('should match the snapshot', () => {
      expect(login).toMatchSnapshot();
    });

    it('should have a default state', () => {
      const expectedDefault = {
        name: '', 
        email: '', 
        password: ''
      };

      expect(login.state()).toEqual(expectedDefault);
    });

    it('should update state based on input changes', () => {
      const mockNameEvent = 
        {target: {value: 'name value', name: 'name'}};
      const mockEmailEvent = 
        {target: {value: 'email value', name: 'email'}};
      const mockPasswordEvent = 
        {target: {value: 'password value', name: 'password'}};
    
      login.instance().handleInputChange(mockNameEvent);
      login.instance().handleInputChange(mockEmailEvent);
      login.instance().handleInputChange(mockPasswordEvent);

      const expectedState = {
        name: 'name value',
        email: 'email value',
        password: 'password value'
      };

      expect(login.state()).toEqual(expectedState);
    });

    it('should call userSignupAttempt on form submit', () => {
      login.instance().submitLogin();
      expect(mockSignupAttempt).toHaveBeenCalled();
    });

    it('should reset input values on form submit', () => {
      const expectedDefault = {
        name: '', 
        email: '', 
        password: ''
      };

      const mockNameEvent = 
        {target: {value: 'name value', name: 'name'}};
      const mockEmailEvent = 
        {target: {value: 'email value', name: 'email'}};
      const mockPasswordEvent = 
        {target: {value: 'password value', name: 'password'}};
    
      login.instance().handleInputChange(mockNameEvent);
      login.instance().handleInputChange(mockEmailEvent);
      login.instance().handleInputChange(mockPasswordEvent);

      login.instance().submitLogin();

      expect(login.state()).toEqual(expectedDefault);
    });
  });

  describe('mapStateToProps', () => {
    let mockStore;
    let result;

    beforeAll(() => {
      mockStore = {
        user: {
          info: {
            name: "Katie"
          }
        },
        error: {
          status: false,
          message: "Error Message Here"
        }
      };
      result = mapStateToProps(mockStore);
    });

    it('should pull user name from the store', () => {
      expect(result.name).toEqual(mockStore.user.info.name);
    });

    it('should pull error status from the store', () => {
      expect(result.errorStatus).toEqual(mockStore.error.status);
    });

    it('should pull error message from the store', () => {
      expect(result.errorMessage).toEqual(mockStore.error.message);
    });
  });

  describe('mapDispatchToProps', () => {
    let mockDispatch;
    let result;

    beforeEach(() => {
      mockDispatch = jest.fn();
      result = mapDispatchToProps(mockDispatch);
    });

    it('should call dispatch when userLoginAttempt is called', () => {
      const mockUserInfo = {mock: 'info'};

      result.userLoginAttempt(mockUserInfo);
      expect(mockDispatch).toHaveBeenCalled();
    });

    it('should call dispatch when userSignupAttempt is called', () => {
      const mockUserInfo = {mock: 'info'};

      result.userSignupAttempt(mockUserInfo);
      expect(mockDispatch).toHaveBeenCalled();
    });

    it('should call dispatch when getCurrentUser is called', () => {
      const mockUserInfo = {mock: 'info'};

      result.getCurrentUser(mockUserInfo);
      expect(mockDispatch).toHaveBeenCalled();
    });
  });
});

