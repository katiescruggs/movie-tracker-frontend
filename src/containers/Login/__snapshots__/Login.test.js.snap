// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Container Login should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    errorMessage="an error message here"
    errorStatus={false}
    getCurrentUser={[Function]}
    name="Katie"
    showRegister={true}
    userLoginAttempt={[Function]}
    userSignupAttempt={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="modal-title"
>
          Sign Up
</h3>,
        <Link
          replace={false}
          to="/"
>
          <button
                    className="btn-close"
          >
                    <FaClose />
          </button>
</Link>,
        null,
        <input
          className="input-field"
          name="name"
          onChange={[Function]}
          placeholder="Name"
          type="text"
          value=""
/>,
        <input
          className="input-field"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          type="email"
          value=""
/>,
        <input
          className="input-field"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
/>,
        <button
          className="btn-submit"
          onClick={[Function]}
>
          Submit
</button>,
      ],
      "className": "login",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign Up",
          "className": "modal-title",
        },
        "ref": null,
        "rendered": "Sign Up",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button
            className="btn-close"
>
            <FaClose />
</button>,
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FaClose />,
            "className": "btn-close",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "button",
        },
        "type": [Function],
      },
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input-field",
          "name": "name",
          "onChange": [Function],
          "placeholder": "Name",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input-field",
          "name": "email",
          "onChange": [Function],
          "placeholder": "Email",
          "type": "email",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input-field",
          "name": "password",
          "onChange": [Function],
          "placeholder": "Password",
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit",
          "className": "btn-submit",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Submit",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="modal-title"
>
            Sign Up
</h3>,
          <Link
            replace={false}
            to="/"
>
            <button
                        className="btn-close"
            >
                        <FaClose />
            </button>
</Link>,
          null,
          <input
            className="input-field"
            name="name"
            onChange={[Function]}
            placeholder="Name"
            type="text"
            value=""
/>,
          <input
            className="input-field"
            name="email"
            onChange={[Function]}
            placeholder="Email"
            type="email"
            value=""
/>,
          <input
            className="input-field"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
/>,
          <button
            className="btn-submit"
            onClick={[Function]}
>
            Submit
</button>,
        ],
        "className": "login",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Up",
            "className": "modal-title",
          },
          "ref": null,
          "rendered": "Sign Up",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="btn-close"
>
              <FaClose />
</button>,
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FaClose />,
              "className": "btn-close",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "button",
          },
          "type": [Function],
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input-field",
            "name": "name",
            "onChange": [Function],
            "placeholder": "Name",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input-field",
            "name": "email",
            "onChange": [Function],
            "placeholder": "Email",
            "type": "email",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input-field",
            "name": "password",
            "onChange": [Function],
            "placeholder": "Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
            "className": "btn-submit",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
